src/main/java/uk/gov/hmcts/ccd/domain/casestate/EnablingConditionSorter.java

```diff
@@ -0,0 +1,29 @@
+package uk.gov.hmcts.ccd.domain.casestate;
+
+import java.util.Comparator;
+import java.util.List;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.domain.model.definition.EventPostStateDefinition;
+
+@Component
+public class EnablingConditionSorter {
+
+    public void sortEventPostStates(List<EventPostStateDefinition> eventPostStates) {
+        if (eventPostStates != null) {
+            eventPostStates.sort(new PostStateComparator());
+        }
+    }
+
+    private class PostStateComparator implements Comparator<EventPostStateDefinition> {
+
+        @Override
+        public int compare(EventPostStateDefinition first, EventPostStateDefinition second) {
+            if (first.getPriority() != null) {
```

Comment:
for my knowledge do we have validation rules on the definition import mandating the priority to be mandatory for a condition? or it can be not specified?

Reply:
Yes, Priority is going to be mandator.. I have removed these checks now